instrumentation/dubbo-rpc/src/main/java/brave/dubbo/rpc/DubboParser.java

```diff
@@ -52,4 +57,37 @@
     String service = url.getServiceInterface();
     return service != null && !service.isEmpty() ? service : null;
   }
+
+  static boolean parseRemoteIpAndPort(Span span) {
+    RpcContext rpcContext = RpcContext.getContext();
+    InetSocketAddress remoteAddress = rpcContext.getRemoteAddress();
+    if (remoteAddress == null) return false;
+    return span.remoteIpAndPort(
+      Platform.get().getHostString(remoteAddress),
+      remoteAddress.getPort()
+    );
+  }
+
+  @Nullable static String errorCode(Throwable error) {
+    if (error instanceof RpcException) {
+      int code = ((RpcException) error).getCode();
+      switch (code) { // requires maintenance if constants are updated
+        case RpcException.UNKNOWN_EXCEPTION:
+          return "UNKNOWN_EXCEPTION";
+        case RpcException.NETWORK_EXCEPTION:
+          return "NETWORK_EXCEPTION";
+        case RpcException.TIMEOUT_EXCEPTION:
+          return "TIMEOUT_EXCEPTION";
+        case RpcException.BIZ_EXCEPTION:
+          return "BIZ_EXCEPTION";
+        case RpcException.FORBIDDEN_EXCEPTION:
+          return "FORBIDDEN_EXCEPTION";
+        case RpcException.SERIALIZATION_EXCEPTION:
+          return "SERIALIZATION_EXCEPTION";
+        default:
+          return String.valueOf(code);
```

Comment:
this is to marry up with rpc.error_code which is defined as a string

Reply:
I wouldn't add anything they don't define as a code as a code, as that is imho more confusing. the fallback to numeric is only there to help us understand drift on dubbo.

Reply:
but yeah if you want I can revert this, just we need it anyway in dubbo for the RPC model, we literally have the string names and should use them.

Reply:
Nah think it's ok

Reply:
probably the _EXCEPTION thing is a bit much I was trying to match what they call it. for example, they have constants and boolean methods for each code http://dubbo.apache.org/docs/2.7.x/api/org/apache/dubbo/rpc/RpcException.html

Reply:
anyway I don't care enough to debate this I'll just put number and it can stay like that even when we move to rpc model.

Reply:
I added them back though I feel really bad about this and think we are doing ourselves and users a disservice. we intentionally say we prefer code names instead of numbers, and we  aren't doing that and offered bad choices instead from an irrelevant comparison with http. I'm not in a convincing mood, so I'll fold despite feeling awful about it.

Reply:
https://github.com/openzipkin/brave/pull/1169#discussion_r412725175

Ah - I meant it seems ok to stick with string too like before the latest commit.

Reply:
I'll resurrect this in a new commit as it is probably deserving to have in a PR title anyway. thanks!